internal/provider/resources/install/aws/aws_staged.go

```diff
@@ -0,0 +1,226 @@
+	State            string `json:"state"`
+}
+
+type StagedAwsComponentJson = struct {
+	Label *string `json:"label"`
+	State *string `json:"state"`
+}
+
+type StagedAwsJson struct {
+	Config struct {
+		Base      map[string]StagedAwsBaseJson      `json:"base"`
+		IamWrite  map[string]StagedAwsComponentJson `json:"iam-write"`
+		Inventory map[string]StagedAwsComponentJson `json:"inventory"`
+	} `json:"config"`
+}
+
+type StagedAwsModel struct {
+	Id               string       `tfsdk:"id"`
+	Label            types.String `tfsdk:"label"`
+	ServiceAccountId types.String `tfsdk:"service_account_id"`
+	Components       []string     `tfsdk:"components"`
+}
+
+func (r *StagedAws) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_aws_staged"
+}
+
+func (r *StagedAws) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: `A staged AWS installation. Staged resources are used to generate AWS trust policies.
+
+**Important** Before using this resource, please read the instructions for the 'aws_iam_write' resource.
+`,
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Required:            true,
+				MarkdownDescription: `The AWS account ID`,
+				Validators: []validator.String{
+					stringvalidator.RegexMatches(AwsAccountIdRegex, "AWS account IDs should be numeric"),
+				},
+			},
+			"label": schema.StringAttribute{
+				Computed:            true,
+				MarkdownDescription: `The AWS account's alias (if available)`,
+			},
+			"service_account_id": schema.StringAttribute{
+				Computed:            true,
+				MarkdownDescription: `The audience ID of the service account to include in this AWS account's P0 role trust policies`,
+			},
+			"components": schema.SetAttribute{
+				ElementType:         types.StringType,
+				Required:            true,
+				MarkdownDescription: `Components to install (any of "iam-write", "inventory")`,
+				Validators: []validator.Set{
+					setvalidator.ValueStringsAre(
+						stringvalidator.OneOf(Components...),
+					),
+					setvalidator.SizeAtLeast(1),
+				},
+			},
+		},
+	}
+}
+
+func (r *StagedAws) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	data := internal.Configure(&req, resp)
+	if data != nil {
+		r.data = data
+	}
+}
+
+func (r *StagedAws) itemPath(component string, id string) string {
+	return fmt.Sprintf("integrations/aws/config/%s/%s", component, id)
+}
+
+func (r *StagedAws) fromComponentJson(data *StagedAwsModel, json *StagedAwsComponentJson, component string) {
+	if json.Label != nil {
+		data.Label = types.StringValue(*json.Label)
+	}
+	data.Components = append(data.Components, component)
+}
+
+func (r *StagedAws) fromJson(data *StagedAwsModel, json *StagedAwsJson) {
```

Comment:
I don't think we need a receiver here.

```suggestion
func fromJson(data *StagedAwsModel, json *StagedAwsJson) {
```

And we can also remove the receiver from `fromComponentJson()`

We also probably shouldn't be taking in a pointer for the source object `json` which doesn't get modified in this function. This would also protect us from accidentally mutating its state unintentionally.

Reply:
I did this just to namespace the `fromJson` function; without this you'll get a clash with similarly named functions for other resources in the same package.

Theoretically you could just define this inline in the Install struct; not sure what the best practice is here.

Reply:
I think normally you can create a separate struct with all the properties but this works fine here 👍 

```
type SomeOtherStruct struct {}

func (r *SomeOtherStruct) fromJson(data *StagedAwsModel, json *StagedAwsJson) {}
```